seqsender.py

```diff
@@ -115,15 +132,20 @@ def update_csv(unique_name,config,type,Genbank_submission_id=None,Genbank_submis
                     "GISAID_failed_total":GISAID_failed_total
                     }
         df = pd.concat([df, pd.Series([new_entry])], ignore_index = True)
-    df.to_csv(os.path.join(os.path.dirname(os.path.abspath(__file__)), "upload_log.csv"), header = True, index = False, sep = ",")
+    df.to_csv(upload_log_path, header = True, index = False, sep = ",")
 
 #Update log status
 #Pulls all entries that do not say processed and updates status
-def update_log():
-    if os.path.isfile(os.path.join(os.path.dirname(os.path.abspath(__file__)), "upload_log.csv")):
-        main_df = pd.read_csv(os.path.join(os.path.dirname(os.path.abspath(__file__)), "upload_log.csv"), header = 0, dtype = str, sep = ",")
+def update_log(upload_log_path: str):
+    def _read_upload_log():
+        return pd.read_csv(upload_log_path, header = 0, dtype = str, sep = ",")
```

Comment:
What's the thinking for closing over the path variable vs taking it as a param and defining this helper outside this function? Is it a way of changing the original seqsender code as little as possible?

This function looks pretty hairy as is. Once we start changing it and taking responsibility for its behaviour, maybe we should be prepared to refactor it entirely.